### PR TITLE
Feature/inspector component section

### PR DESCRIPTION
**Pull request by OmarDaSilva**
Added feature to the component section. Depending on the currently selected component, it will display different kinds of information on the inspector depending if it is an NPM imported component, focusable component, or non-focusable component. Non-npm components have an icon button to toggle focus.

Screenshot 2021-04-26 at 14 01 59

Screenshot 2021-05-12 at 15 49 30

Screenshot 2021-05-12 at 15 50 11

### CODE REVIEW

**Comment by OmarDaSilva** [comment]
Left todo: wire in the if statements

**Comment by OmarDaSilva** [comment]
Ready for review!

**Comment by maltenuhn** [comment]
Nice!

A couple of things i noticed:
- there's some rogue $ characters in there. `${variableName}` only works if it's enclosed in back ticks (`). See [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) for how that works
- `${elementName}` isn't the component type ("Styled Component", "Animated Component" etc) - not sure if we have that available yet

**Comment by OmarDaSilva** [comment]
This branch is now ready for review

**Review by seanparsons — APPROVED** [review]

**Comment by OmarDaSilva** [comment]
This PR is now ready for feedback

File: editor/src/components/inspector/sections/component-section/component-section.tsx [hunk 1]
```diff
@@ -514,11 +567,91 @@ export const ComponentSectionInner = betterReactMemo(
...
+      return {
+        editor: store.editor,
+        derived: store.derived,
+      }
+    }, 'Focusable values')
+
+    const target = selectedViews[0]
+    // const pathAsScenePath = EP.isScenePath(target)
+    //   ? target
+    //   : EP.scenePathForElementAtInstancePath(target)
```
Comment (Rheeseyb) [inline editor/src/components/inspector/sections/component-section/component-section.tsx:586]:
```suggestion
```

File: editor/src/components/inspector/sections/component-section/component-section.tsx [hunk 2]
```diff
@@ -514,11 +567,91 @@ export const ComponentSectionInner = betterReactMemo(
...
+
+    const underlyingTarget = normalisePathToUnderlyingTarget(
+      editor.projectContents,
+      editor.nodeModules.files,
+      editor.canvas.openFile?.filename ?? '',
+      selectedViews[0],
+    )
+    const locationOfComponentInstance =
+      underlyingTarget.type === 'NORMALISE_PATH_SUCCESS' ? underlyingTarget.filePath : ''
+    // you can write isImportedComponent, isEmotionOrStyledComponent -> npm logo or not
```
Comment (Rheeseyb) [inline editor/src/components/inspector/sections/component-section/component-section.tsx:625]:
```suggestion
```
?

I'm assuming this was a note to yourself whilst working on this?

File: editor/src/components/inspector/sections/component-section/component-section.tsx [hunk 3]
```diff
@@ -529,21 +662,28 @@ export const ComponentSectionInner = betterReactMemo(
...
-                <SimpleFlexRow style={{ flexGrow: 1 }}>Component props</SimpleFlexRow>
-                <SquareButton highlight onClick={onResetClicked}>
+                <UIGridRow
+                  padded
+                  variant='|--32px--|<--------auto-------->'
+                  style={{ flexGrow: 1, color: colorTheme.primary.value }}
+                >
+                  Component
+                </UIGridRow>
+                {/* <SquareButton highlight onClick={onResetClicked}>
```
Comment (Rheeseyb) [inline editor/src/components/inspector/sections/component-section/component-section.tsx:672]: Is this intentional? If so please delete.

**Review by Rheeseyb — APPROVED** [review]

**Comment by balazsbajorics** [comment]
could you post new screenshots please? :)

**Pull request merged**